python/pyarrow/parquet/core.py

```diff
@@ -2081,11 +2081,11 @@ def file_visitor(written_file):
         the entire directory will be deleted.  This allows you to overwrite
         old partitions completely.
     **kwargs : dict,
-        Used as additional kwargs for :func:`pyarrow.dataset.write_dataset`
+        Used as additional kwargs for :py:func:`pyarrow.dataset.write_dataset`
```

Comment:
`py` is indeed the default domain (unless [configured](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-primary_domain)  otherwise). I have however the habit to keep it explicit, mainly because I feel it's better [documented in sphinx docs](https://www.sphinx-doc.org/en/master/usage/domains/python.html#python-xref-roles) and easier to google for newcomers I guess.

But again, I'm happy to drop it again to better align with guidelines